src/mame/drivers/fmtowns.cpp

```diff
@@ -2960,6 +2973,15 @@ void towns_state::towns(machine_config &config)
 {
 	towns_base(config);
 	NVRAM(config, "nvram", nvram_device::DEFAULT_ALL_0);
+
+	FMT_SCSI_SLOT(config, m_scsi_slot, fmt_scsi_default_devices, nullptr);
+	m_scsi_slot->irq_handler().set(FUNC(towns_state::towns_scsi_irq));
+	m_scsi_slot->drq_handler().set(FUNC(towns_state::towns_scsi_drq));
+
+	m_dma[0]->dma_read_callback<1>().set(FUNC(towns_state::towns_scsi_dma_r));
+	m_dma[0]->dma_write_callback<1>().set(FUNC(towns_state::towns_scsi_dma_w));
+	m_dma[1]->dma_read_callback<1>().set(FUNC(towns_state::towns_scsi_dma_r));
+	m_dma[1]->dma_write_callback<1>().set(FUNC(towns_state::towns_scsi_dma_w));
```

Comment:
You know you have a slot and not an integrated SCSI controller here.  You should bind to it directly and avoid the trampoline with its conditional jump.  (The trampoline can probably be avoided for the systems with an integrated SCSI controller, too.  It’s likely an artefact of the days of `MCFG_` soup when machine configuration was a black art.)

Reply:
Done. Like you said, I was able to get rid of the trampolines entirely, even for the integrated controller.